File: 2023/UICPC Regional/Round #15/solutions/hothike/per.ml
let n = int_of_string(read_line());;
let x = List.map int_of_string (String.split_on_char ' ' (read_line()));;
let rec solve i =
  if i+2 >= n then (50, 0)
  else min (max (List.nth x i) (List.nth x (i+2)), i+1) (solve (i+1));;
Printf.printf "%d %d\n" (snd (solve 0)) (fst (solve 0));;

File: 2023/UICPC Regional/Round #15/solutions/buildingboundaries/per.py
def solve():
    L = map(int, raw_input().split())
    X = zip([0,1,2]*2, L[::2]+L[1::2], L[1::2]+L[::2])
    print min(min(max(a1+a2,a3)*(b1+b3), (a1+a2+a3)*max(b1,b2,b3), max(a1+a2,a1+a3)*max(b1,b2+b3))
              for (i,a1,b1) in X for (j,a2,b2) in X for (k,a3,b3) in X
              if i != j and i+j+k == 3 and b2 <= b1)

for _ in range(int(raw_input())): solve()
